syntax = "proto3";
import "google/protobuf/empty.proto";
import "indevicecommunication.proto";

option java_multiple_files = true;
option objc_class_prefix = "HLW";

package controlmessages;

service ControlMessages {
  rpc AdvertiseToController(ConnectionConfigs) returns (SystemInfo) {}
  rpc SendDummyData(DummyMessage) returns (google.protobuf.Empty) {}
  rpc ForwardFlRequest(indevicecommunication.FlData) returns (google.protobuf.Empty) {}
}

message ConnectionConfigs {
  string device_name = 1;
  uint32 device_type = 2;
  string ip_address = 3;
  int32 processors = 4;
  repeated uint64 memory = 5;
}

message SystemInfo {
  string name = 1;
  string experiment = 2;
}

message DummyMessage {
  string origin_name = 1;
  int64 gen_time = 2;
  bytes data = 3;
}